src/test/kotlin/com/codingchallenge/transactionanalyser/TransactionLoaderTest.kt

```diff
@@ -0,0 +1,52 @@
+package com.codingchallenge.transactionanalyser
+
+import com.codingchallenge.transactionanalyser.domain.TransactionType
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import java.math.BigDecimal
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+internal class TransactionLoaderTest{
+    private val transactionLoader = TransactionLoader(
+        this::class.java.classLoader.getResourceAsStream("transactions.csv") ?: throw IllegalArgumentException("File not found")
+    )
+    private val formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss")
+
+    @Test
+    fun givenCsv_WhenLoadTransactionsFromCsv_ThenTheExpectedTransactionsReturned(){
+        val transactions = transactionLoader.loadTransactionsFromCsv()
+        assertEquals(transactions.size, 5);
+
+        assertEquals(transactions[0].transactionId, "TX10001")
+        assertEquals(transactions.get(0).fromAccountId, "ACC334455")
+        assertEquals(transactions[0].transactionId, "TX10001")
+        assertEquals(transactions[0].fromAccountId, "ACC334455")
+        assertEquals(transactions[0].toAccountId, "ACC778899")
+        assertEquals(transactions[0].createdAt, LocalDateTime.parse("20/10/2018 12:47:55", formatter))
+        assertEquals(transactions[0].amount.compareTo(BigDecimal.valueOf(25.00)), 0)
+        assertEquals(transactions[0].transactionType, TransactionType.PAYMENT)
+        assertNull(transactions[0].relatedTransaction) // PAYMENT transaction should have null relatedTransaction
+
+
+        assertEquals(transactions[1].transactionId, "TX10002")
+        assertEquals(transactions[1].fromAccountId, "ACC334455")
+        assertEquals(transactions[1].toAccountId, "ACC998877")
+        assertEquals(transactions[1].createdAt, LocalDateTime.parse("20/10/2018 17:33:43", formatter))
+        assertEquals(transactions[1].amount.compareTo(BigDecimal.valueOf(10.50)), 0)
+        assertEquals(transactions[1].transactionType, TransactionType.PAYMENT)
+        assertNull(transactions[1].relatedTransaction)
+
+        assertEquals(transactions[3].transactionId, "TX10004")
+        assertEquals(transactions[3].fromAccountId, "ACC334455")
+        assertEquals(transactions[3].toAccountId, "ACC998877")
+        assertEquals(transactions[3].createdAt, LocalDateTime.parse("20/10/2018 19:45:00", formatter))
+        assertEquals(transactions[3].amount.compareTo(BigDecimal.valueOf(10.50)), 0)
+        assertEquals(transactions[3].transactionType, TransactionType.REVERSAL)
```

Comment:
there are different schools of thought on this, but generally I believe one test should test one thing.
If something in the implementation changes, your test are there to cover you. Every time you make a small change to the implementation, you run the tests, they pass, you are ok. But if one thing changes in our implementation and the test fails and that test is doing **soooo** many things, we almost then need to go digging through the test itself to figure out where in the test it broke as well.
Consider small tests that have a single assert statement (or not more than one or two, fewer is better than lots). If a change is made to your implementation and a test breaks, that test should be very clear on what exactly went wrong. Don't make more work for yourself just because it's easier to add lots of assert statements in one place; treat the Clean Code principles equal in your test code as in your implementation code. It will serve you well.

Reply:
In this case we have 3 different transactions being tested all at once. At the very least, we could assert differently for those. 
In addition, perhaps we could have one assert statement on a single transaction object, rather than for every field one by one on that transaction

Reply:
Agree it should be broken down into smaller test methods. I was being lazy.